src/erl_optics_server.erl

```diff
@@ -0,0 +1,120 @@
+-module(erl_optics_server).
+-include("erl_optics.hrl").
+
+-behaviour(gen_server).
+
+-export([init/1,
+         handle_call/3,
+         handle_cast/2,
+         handle_info/2,
+         terminate/2,
+         code_change/3]).
+
+-export([start_link/0,
+         stop/0]).
+
+%for prototyping only
+-export([start/0,
+         test_update/0,
+         start_test/1]).
+
+-define(SERVER, ?MODULE).
+
+-record(state, {port  :: undefined | non_neg_integer(),
+                addr  :: undefined | list(),
+                mode  :: undefined | carbon | prometheus | blank}).
+
+
+%%%=========
+%%% API
+%%%=========
+
+%Modes: prometheus | carbon | blank
+
+start_link() ->
+    gen_server:start_link({local, ?SERVER}, ?MODULE, [], []).
+
+stop() ->
+    gen_server:cast(?SERVER, stop).
+
+start_test(Interval) ->
+    gen_server:cast(?SERVER, {test_update, Interval}).
+
+%%%==========
+%%% Callbacks
+%%%==========
+
+init([]) ->
+    Mode = ?ENV(?ENV_MODE, ?DEFAULT_MODE),
+    case Mode of
+        carbon ->
+            Hostname = ?ENV(?ENV_HOSTNAME, ?DEFAULT_HOSTNAME),
+            Port = ?ENV(?ENV_PORT, ?DEFAULT_PORT),
+            Interval = ?ENV(?ENV_INTERVAL, ?DEFAULT_INTERVAL),
+            erl_optics:register_carbon_poller(Hostname, Port),
+            timer:send_interval(Interval, carbon_poll),
+            %gen_server:cast(?SERVER, {test_update, 10}), %for testing
+            {ok, #state{mode = carbon, port = Port, addr = Hostname}, 0};
+        prometheus ->
+            %to be implemented
+            {ok, #state{mode = prometheus}, 0};
+        blank ->
+            {ok, #state{mode = blank}, 0}
```

Comment:
Why is a timeout specified if the `handle_info` for `timeout` does nothing? If there's no reason for using it, please remove it.